front-end/src/pages/race/RacePage.vue

```diff
@@ -70,31 +75,64 @@ export default defineComponent({
             isRaceDeleteModalVisible.value = true;
         };
 
-        const route = useRoute();
-        let raceId = route.params.id;
-        let raceName = ref<string>('test');
+        let isTextModalVisible = ref(false)
 
-        if (raceId != 'f47ac10b-58cc-4372-a567-0e02b2c3d479') {
-            router.push('/');
-        }
+        let textModalContent = ref<textModal>({
+            header: 'Alert',
+            body: 'Nothing to show'
+        });
+
+        const showModal = (header: string, body: string) => {
+            textModalContent.value.header = header;
+            textModalContent.value.body = body;
+            isTextModalVisible.value = true;
+        };
+
+        const route = useRoute();
+        let raceId = Number(route.params.id);
+        let raceName = ref<string>('');
 
         const navigateTo = (location: string) => {
             router.push(location);
         };
 
         const handleRaceNameChange = async (newRaceName: string) => {
-            raceName.value = newRaceName;
-            isRaceNameChangeModalVisible.value = false;
+            raceService.updateRaceName(raceId, newRaceName)
+                .then(() => {
```

Comment:
deze functies zijn async, dus waarom niet gewoon `await` in plaats van `.then()`?